### PR TITLE
JNG-4208 Run tests with postgres

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4208" title="JNG-4208" target="_blank">Sub-taskJNG-4208</a>  Run test in judo-runtime-core-jsl CI
  </td></table>
  <br />
 

This PR contains the following changes:
- maven build on CI separated into:
    - full build with HSQLDB tests
    - tests with PostgreSQL
    - deploy to judong
- commented source code removed from pom
- `JudoDatasourceByClassExtension`, `JudoDatasourceFixture`, `JudoDatasourceSingletonExtension` and `YugabytedbSQLContainer` java classes copied from esm-itest to enable multi-dialect testing
- `AbstractJslTest` completed with separate module loading for HSQLDB and PostgreSQL dialects
- `AbstractJslTest` completed with flat transaction management "support"
- `@Disabled` annotations completed with referenced tickets
- tests updated according to fixed numeric result cutting (JNG-4292 included)
- invalid assertions (with certain dialects) temporarily commented
- assertion related to JNG-4179 re-enabled with correction
- max precision test fixed
- `maven-install-plugin` configuration added to enable deploy only mode
- dependencies added for PostgreSQL testing

Removed changes:
- ~test source generation in jsl-itest wrapped in a profile to be able to skip those if sources are already present for testing~
- ~`@Disabled` annotations completed with // TODO part to be able find disabled tests with more search criteria~
- ~Assertions with different results on different dialects separated~
- ~deploy-only property name changed to follow naming conventions~

### CODE REVIEW

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java [hunk 1]
```diff
@@ -285,28 +280,29 @@ void testUpdateRootUpdatesInterimElement() {
...
-        TransactionStatus transactionStatus = transactionManager.getTransaction(new DefaultTransactionDefinition());
+    void testManualTransactionManagementCommit() {
+        beginTransaction();
         assertEquals(Optional.of("TEST-A"), entityADao.getById(entityA.get__identifier()).get().getStringA());
         entityA.setStringA("BLAAA");
         entityADao.update(entityA);
-        transactionManager.commit(transactionStatus);
+        commitTransaction(); // TODO: if this change is committed, then afterEach won't be able to rollback to enable clean state for the other tests
         assertEquals(Optional.of("BLAAA"), entityADao.getById(entityA.get__identifier()).get().getStringA());
     }
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: TBD

Comment (noherczeg) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: ?

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java [hunk 2]
```diff
@@ -131,12 +132,15 @@ void testOperatorsForDefaultValues() {
         assertEquals(Optional.of(true), operators.getStringEq());
         assertEquals(Optional.of(true), operators.getStringNeq());
 
-        assertEquals(Optional.of(true), operators.getCsStringLt());
+        // TODO: inconsistently consistent string comparison, https://blackbelt.atlassian.net/browse/JNG-4045
+        if (DIALECT_HSQLDB.equals(dialect)) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java]: We don't do this here.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java [hunk 3]
```diff
@@ -131,12 +132,15 @@ void testOperatorsForDefaultValues() {
...
+            assertEquals(Optional.of(true), operators.getCsStringLt());
+            assertEquals(Optional.of(true), operators.getCsStringLte2());
+            assertEquals(Optional.of(true), operators.getCsStringGt());
+            assertEquals(Optional.of(true), operators.getCsStringGte2());
+        }
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java]: Please make a ticket and a disabled Test for it. Do not make condition in test for dialect.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java [hunk 4]
```diff
@@ -285,8 +289,16 @@ public void testStrings() {
...
+
+        // TODO: https://blackbelt.atlassian.net/browse/JNG-4293
+        if (DIALECT_POSTGRESQL.equals(dialect)) {
+            assertEquals("ap", str.getLpad2().orElseThrow());
+            assertEquals("ap", str.getLpad3().orElseThrow());
+        } else {
+            assertEquals("le", str.getLpad2().orElseThrow());
+            assertEquals("le", str.getLpad3().orElseThrow());
+        }
+
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: Please move to tests where the test is disabled and give the ticket number.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java [hunk 5]
```diff
@@ -494,8 +506,11 @@ public void testTime() {
         assertEquals(Optional.of(59L), time.getSecond());
         assertEquals(LocalTime.of(13, 45, 0), time.getOf().orElseThrow());
 
-        assertEquals(LocalTime.of(11, 11, 11), time.getTimeFromSeconds().orElseThrow());
-        assertEquals(40271L, time.getTimeAsSeconds().orElseThrow());
+        // TODO: https://blackbelt.atlassian.net/browse/JNG-1586, https://blackbelt.atlassian.net/browse/JNG-3681
+        if (DIALECT_HSQLDB.equals(dialect)) {
+            assertEquals(LocalTime.of(11, 11, 11), time.getTimeFromSeconds().orElseThrow());
+            assertEquals(40271L, time.getTimeAsSeconds().orElseThrow());
+        }
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: Please move to tests where the test is disabled and give the ticket number.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -99,22 +99,37 @@ jobs:
...
+          -f judo-runtime-core-jsl-itest \
+          test
+
+      - name: 🎁 Deploy with Maven to judong
+        run: |-
+          ./mvnw -B -Dstyle.color=always \
+          -Drevision=${{ steps.version.outputs.version }} \
+          -Ddeploy-only \
+          -Dmaven.test.skip=true \
+          -P"sign-artifacts,release-judong" \
```
Comment (robertcsakany) [inline .github/workflows/build.yml]: Have to avoid this hack.

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -99,22 +99,37 @@ jobs:
...
 
-      # Phase 2 - Build and deploy to maven central
-      - name: 🎁 Deploy with Maven (central)
+      - name: 🎁 Deploy with Maven to central
         if: ${{ always() && startsWith(github.head_ref, 'release') && job.status == 'success' }}
         run: |-
           ./mvnw -B -Dstyle.color=always \
           -Drevision=${{ steps.version.outputs.version }} \
-          -DdeployOnly \
+          -Ddeploy-only \
```
Comment (robertcsakany) [inline .github/workflows/build.yml]: There is no benefit of this change.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java (hunk 3 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/OperatorsTest.java]: [a687597](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/a687597c3bb5057613402deac3fc75a395a45ce6)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java (hunk 4 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: [a687597](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/a687597c3bb5057613402deac3fc75a395a45ce6)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java (hunk 5 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: [a687597](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/a687597c3bb5057613402deac3fc75a395a45ce6)

File: .github/workflows/build.yml (hunk 6 above)
Comment (bencegelei) [reply .github/workflows/build.yml]: [a56f6eb](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/a56f6eb4c92d995a97574b0f499d0149c997cf08) [a20db73](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/a20db73e0cbfb3228bd0b7c98a68aa3d75789b4d)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java (hunk 1 above)
Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: TODO: separate TransactionManagementTest

Comment (bencegelei) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: [f2d74e5](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/f2d74e5f661057a103d83891c6ada91d3870f341)

File: .github/workflows/build.yml (hunk 7 above)
Comment (bencegelei) [reply .github/workflows/build.yml]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/97/commits/c317dba5ba16296798697565ccb3a31570f298d2

**Pull request merged**